gstreamer-sdp/src/auto/s_d_p_media.rs

```diff
@@ -0,0 +1,260 @@
+// This file was generated by gir (0113735) from gir-files (???)
+// DO NOT EDIT
+
+use SDPResult;
+use ffi;
+use glib::translate::*;
+use glib_ffi;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct SDPMedia(Boxed<ffi::GstSDPMedia>);
+
+    match fn {
+        copy => |ptr| ffi::gst_sdp_media_copy(mut_override(ptr)),
+        free => |ptr| ffi::gst_sdp_media_free(ptr),
+    }
+}
+
+impl SDPMedia {
+    pub fn add_attribute(&mut self, key: &str, value: &str) -> SDPResult {
+        unsafe {
+            from_glib(ffi::gst_sdp_media_add_attribute(self.to_glib_none_mut().0, key.to_glib_none().0, value.to_glib_none().0))
+        }
+    }
+
+    pub fn add_bandwidth(&mut self, bwtype: &str, bandwidth: u32) -> SDPResult {
+        unsafe {
+            from_glib(ffi::gst_sdp_media_add_bandwidth(self.to_glib_none_mut().0, bwtype.to_glib_none().0, bandwidth))
+        }
+    }
+
+    pub fn add_connection(&mut self, nettype: &str, addrtype: &str, address: &str, ttl: u32, addr_number: u32) -> SDPResult {
+        unsafe {
+            from_glib(ffi::gst_sdp_media_add_connection(self.to_glib_none_mut().0, nettype.to_glib_none().0, addrtype.to_glib_none().0, address.to_glib_none().0, ttl, addr_number))
+        }
+    }
+
+    pub fn add_format(&mut self, format: &str) -> SDPResult {
+        unsafe {
+            from_glib(ffi::gst_sdp_media_add_format(self.to_glib_none_mut().0, format.to_glib_none().0))
+        }
+    }
+
+    pub fn as_text(&self) -> Option<String> {
+        unsafe {
+            from_glib_full(ffi::gst_sdp_media_as_text(self.to_glib_none().0))
+        }
+    }
+
+    pub fn attributes_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_attributes_len(self.to_glib_none().0)
+        }
+    }
+
+    //pub fn attributes_to_caps(&self, caps: /*Ignored*/&mut gst::Caps) -> SDPResult {
```

Comment:
Needs to be manual and use `&mut CapsRef`

Reply:
Not generated for now